File: Python/011.py
class Solution(object):
    def maxArea(self, hight):
        """
        :type hights: List[int]
        :rtype: int
        """
        hights=[]
        for i in range(1, len(hight)):
            if hight[i] > hight[i-1]:
                hights.append(hight[i - 1])
            else:
                hights.append(hight[i])

        hmin=min(hights)
        hmax=max(hights)
        maxs= len(hights) * hmin
        for i in range(hmin+1,hmax+1):
            wight=0
            flag=1
            for temp in hights:
                if temp>=i:
                    print "hi",
                    if flag == 0:
                        wight=wight+1
                    else:
                        flag=0
                        wight=1
                    print wight
                else:
                    area=wight*i
                    print area,wight,i
                    if area > maxs:
                        maxs=area
                    flag=1
            area = wight * i
            print area, wight, i
            if area > maxs:
                maxs = area
        return maxs

if __name__ == '__main__':
    example=Solution()
    height=[1,2,3,4]
    print example.maxArea(height)

File: C++/011.c
#include<stdio.h>
#include<iostream>
using namespace std;
int maxArea(int* ,int);
int maxArea(int* height, int heightSize) {
    int i=0,j=0;
    int temp=0;
    int max=0;
    int area=0;
    int h1=0,h2=0;
    for(temp=0;temp<heightSize;temp++)
    {
        h2=*(height+temp);
        //printf("%d",h2);
        for(i=0;i<heightSize;i++)
        {
            //h3=*(height+i);
            if ( *(height+i) >=h2)
            {
                for(j=heightSize-1;j>i;j--)
                {
                    h1=*(height+j);
                    if (h1>=h2)
                    {
                        area=(j-i)*h2;
                        cout<<h1<<' '<<h2<<' '<<area<<endl;
                        if (area>max)
                            max=area;
                        break;
                    }
                }
                break;
            }
        }
    }
    return max;
}

int main(){
    int height[]={1,2,3,4};
    int result = maxArea(height, 4 );
    cout<<result<<endl;
    
}
